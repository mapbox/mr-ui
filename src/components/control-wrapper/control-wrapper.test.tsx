import React from 'react';
import ControlWrapper from './control-wrapper';
import { render } from '@testing-library/react';

describe('ControlWrapper', () => {
  describe('basic', () => {
    const props = {
      id: 'control-wrapper',
      children: <span>fake input</span>
    };

    test('renders', () => {
      const { baseElement } = render(<ControlWrapper {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('basic with error', () => {
    const props = {
      id: 'control-wrapper',
      validationError: 'There was an error',
      children: <span>fake input</span>,
      themeLabel: 'bg-red'
    };

    test('renders', () => {
      const { baseElement } = render(<ControlWrapper {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('jsx error', () => {
    const props = {
      id: 'control-wrapper',
      validationError: (
        <div>
          <div className="txt-bold mb3">Something went wrong!</div>
          <div>You made a huge mistake</div>
        </div>
      ),
      children: <span>fake input</span>,
      themeLabel: 'bg-red'
    };

    test('renders', () => {
      const { baseElement } = render(<ControlWrapper {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });
});
