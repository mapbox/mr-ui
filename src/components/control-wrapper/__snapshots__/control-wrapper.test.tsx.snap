// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlWrapper basic renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <span>
        fake input
      </span>
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;

exports[`ControlWrapper basic with error renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <span>
        fake input
      </span>
      <div
        role="alert"
      >
        <div
          class="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
          data-testid="control-wrapper-error"
          id="control-wrapper-error"
        >
          There was an error.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ControlWrapper jsx error renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <span>
        fake input
      </span>
      <div
        role="alert"
      >
        <div
          class="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
          data-testid="control-wrapper-error"
          id="control-wrapper-error"
        >
          <div>
            <div
              class="txt-bold mb3"
            >
              Something went wrong!
            </div>
            <div>
              You made a huge mistake
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
